pkg/controller/controller.go

```diff
@@ -354,17 +365,22 @@ func checkDriverCapabilities(grpcClient *grpc.ClientConn, timeout time.Duration,
 		return nil, fmt.Errorf("no create/delete volume support detected")
 	}
 
-	// If PVC.Spec.DataSource is not nil, it indicates the request is to create volume
-	// from snapshot and therefore we should check for snapshot support;
-	// otherwise we don't need to check for snapshot support.
-	if needSnapshotSupport {
+	if caps.snapshot {
 		// Check whether plugin supports create snapshot
 		// If not, create volume from snapshot cannot proceed
 		if !capabilities.Has(ControllerCapability_CREATE_DELETE_SNAPSHOT) {
 			return nil, fmt.Errorf("no create/delete snapshot support detected. Cannot create volume from snapshot")
 		}
 	}
 
+	if caps.clone {
+		// Check whether plugin supports create clone
+		// If not, return error
+		if !capabilities.Has(ControllerCapability_CLONE_VOLUME) {
+			return nil, fmt.Errorf("no create/delete snapshot support detected. Cannot create volume from snapshot")
```

Comment:
The error should be updated
"no clone volume support detected. Cannot create volume from volume"